cmd/gitcloner/integrationtests/clone_test.go

```diff
@@ -0,0 +1,465 @@
+					cloneErr = c.CloneRepo(opts)
+					Expect(cloneErr).NotTo(HaveOccurred())
+				})
+
+				It("README.md file is present", func() {
+					_, err := os.Stat(tmp + "/README.md")
+					Expect(err).NotTo(HaveOccurred())
+				})
+			})
+		})
+
+		When("Insecure skip tls is not true, and no caBundle is provided", func() {
+			BeforeEach(func() {
+				private = false
+				opts = &cmd.Options{}
+			})
+
+			JustBeforeEach(func() {
+				c := gogit.NewCloner()
+				cloneErr = c.CloneRepo(opts)
+			})
+
+			It("Fails to clone the repo", func() {
+				Expect(cloneErr.Error()).To(ContainSubstring("x509: certificate signed by unknown authority"))
+			})
+		})
+
+		When("Insecure skip tls is not true, and caBundle is provided", func() {
+			var tmpFile string
+
+			BeforeEach(func() {
+				private = false
+				opts = &cmd.Options{}
+			})
+
+			AfterEach(func() {
+				err := os.Remove(tmpFile)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			JustBeforeEach(func() {
+				caBundleFile, err := os.CreateTemp("", "gitcloner")
+				Expect(err).NotTo(HaveOccurred())
+				tmpFile = caBundleFile.Name()
+				_, err = caBundleFile.Write(gogsCABundle)
+				Expect(err).NotTo(HaveOccurred())
+				opts.CABundleFile = caBundleFile.Name()
+
+				c := gogit.NewCloner()
+				cloneErr = c.CloneRepo(opts)
+			})
+
+			It("README.md file is present", func() {
+				Expect(cloneErr).NotTo(HaveOccurred())
+				_, err := os.Stat(tmp + "/README.md")
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+	})
+
+	When("Cloning an ssh repo", func() {
+		var tmpKey string
+
+		JustBeforeEach(func() {
+			container, err := createGogsContainer()
+			Expect(err).NotTo(HaveOccurred())
+			url, err := getHTTPSURL(context.Background(), container)
+			Expect(err).NotTo(HaveOccurred())
+			err = createRepo(url, "assets/repo", private)
+			Expect(err).NotTo(HaveOccurred())
+			sshURL, err := getSSHURL(context.Background(), container)
+			Expect(err).NotTo(HaveOccurred())
+			opts.Repo = sshURL + "/test/test-repo"
+			tmp = createTempFolder()
+			opts.Path = tmp
+			// create private key, and register public key in gogs
+			key, err := createAndAddKeys()
+			Expect(err).NotTo(HaveOccurred())
+			tmpFile, err := os.CreateTemp("", "testkey")
+			Expect(err).NotTo(HaveOccurred())
+			_, err = tmpFile.Write([]byte(key))
+			Expect(err).NotTo(HaveOccurred())
+			tmpKey = tmpFile.Name()
+			opts.SSHPrivateKeyFile = tmpKey
+		})
+
+		JustAfterEach(func() {
+			err := os.RemoveAll(tmp)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		When("Private repo that contains a README.md file", func() {
+			BeforeEach(func() {
+				private = true
+				opts = &cmd.Options{
+					InsecureSkipTLS: true,
+				}
+			})
+
+			AfterEach(func() {
+				err := os.RemoveAll(tmpKey)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			JustBeforeEach(func() {
+				c := gogit.NewCloner()
+				cloneErr = c.CloneRepo(opts)
+				Expect(cloneErr).NotTo(HaveOccurred())
+			})
+
+			It("README.md file is present", func() {
```

Comment:
Same comment as on L81.

Reply:
done in https://github.com/rancher/gitjob/commit/43c66e7e40a9791d03bff8bc45172e4d1e87742e